BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonRequest.java

```diff
@@ -36,7 +36,7 @@
   @Deprecated
   private final Optional<String> replaceServiceId;
 
-  private final Optional<RequestAction> action;
+  private Optional<RequestAction> action;
```

Comment:
There is a toBuilder to allow this while keeping the overall object immutable. We've tried to keep them immutable and make copies vs mutating where possible